test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -240,107 +155,133 @@ func schedulePods(config *testConfig) int32 {
 	}
 }
 
-// mutateNodeSpec returns the strategy needed for creation of nodes.
-// TODO: It should take the nodespec and return the modified version of it. As of now, returning the strategies for backward compatibilty.
-func (na nodeAffinity) mutateNodeSpec(numNodes int) []testutils.CountToStrategy {
-	numGroups := na.numGroups
-	nodeAffinityKey := na.nodeAffinityKey
-	nodeStrategies := make([]testutils.CountToStrategy, 0, numGroups)
-	for i := 0; i < numGroups; i++ {
-		nodeStrategies = append(nodeStrategies, testutils.CountToStrategy{
-			Count:    numNodes / numGroups,
-			Strategy: testutils.NewLabelNodePrepareStrategy(nodeAffinityKey, fmt.Sprintf("%v", i)),
-		})
+// getBaseNodeTemplate returns the base node template.
+func getBaseNodeTemplate() *v1.Node {
+	baseNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "sample-node-",
+		},
+		Spec: v1.NodeSpec{
+			// TODO: investigate why this is needed.
+			ExternalID: "foo",
+		},
+		Status: v1.NodeStatus{
+			Capacity: v1.ResourceList{
+				v1.ResourcePods:   *resource.NewQuantity(110, resource.DecimalSI),
+				v1.ResourceCPU:    resource.MustParse("4"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+			Phase: v1.NodeRunning,
+			Conditions: []v1.NodeCondition{
+				{Type: v1.NodeReady, Status: v1.ConditionTrue},
+			},
+		},
 	}
-	return nodeStrategies
+	return baseNode
 }
 
-// mutatePodSpec returns the list of pods after mutating the pod spec based on predicates and priorities.
-// TODO: It should take the podspec and return the modified version of it. As of now, returning the podlist for backward compatibilty.
-func (na nodeAffinity) mutatePodSpec(numPods int, pod *v1.Pod) []*v1.Pod {
-	numGroups := na.numGroups
-	nodeAffinityKey := na.nodeAffinityKey
-	podList := make([]*v1.Pod, 0, numGroups)
-	for i := 0; i < numGroups; i++ {
-		pod = &v1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "sched-perf-node-affinity-pod-",
-			},
-			Spec: testutils.MakePodSpec(),
-		}
-		pod.Spec.Affinity = &v1.Affinity{
-			NodeAffinity: &v1.NodeAffinity{
-				RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
-					NodeSelectorTerms: []v1.NodeSelectorTerm{
-						{
-							MatchExpressions: []v1.NodeSelectorRequirement{
-								{
-									Key:      nodeAffinityKey,
-									Operator: v1.NodeSelectorOpIn,
-									Values:   []string{fmt.Sprintf("%v", i)},
-								},
-							},
-						},
+// getBasePodTemplate returns the base pod template.
+func getBasePodTemplate() *v1.Pod {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "sched-perf-pod-",
+		},
+		Spec: testutils.MakePodSpec(),
+	}
+	return pod
+}
+
+// mutateNode returns the modified node needed for creation of nodes.
+func (na nodeAffinity) mutateNode(node *v1.Node) {
```

Comment:
Yes, every predicate and priority will have this. I tried with taints and it was able to pick up, I will add taints in another PR though.